src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,96 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int MIN_INITIAL_CAPACITY = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75F;
+    static final int MULTIPLIER_FOR_RESIZE = 2;
+    private Node<K, V>[] bucket;
+    private int size = 0;
+    private int threshold;
+
+    public MyHashMap() {
+        this.bucket = new Node[MIN_INITIAL_CAPACITY];
+        this.threshold = (int) (MIN_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size + 1 >= threshold) {
+            resize();
+        }
+        int index = getIndex(key);
+        if (bucket[index] == null) {
+            Node<K, V> node = new Node<>(key, value, null);
+            bucket[index] = node;
+            size++;
+        } else {
+            Node<K, V> currentNode = bucket[index];
+            while (currentNode != null) {
+                if (Objects.equals(currentNode.key, key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                if (currentNode.next == null) {
+                    currentNode.next = new Node<>(key, value, null);
```

Comment:
```suggestion
                    currentNode.next = new Node<>(key, value);
```

Reply:
Create constructor with 2 parameters .The null was deleted .

Reply:
Many thanks!